### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

Achtung!!!

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-generic.git`](https://github.com/cisagov/skeleton-generic.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/cyhy_amis.git cyhy_amis
    cd cyhy_amis
    git remote add skeleton https://github.com/cisagov/skeleton-generic.git
    git remote set-url --push skeleton no_push
    git switch develop
    git switch --create lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .github/dependabot.yml 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may *append* to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Confirm each item in the "Pre-approval checklist" below.

1. Remove any of the checklist items that do not apply.

1. Ensure every remaining checkbox has been checked.

1. Mark this draft pull request "Ready for review".

## ✅ Pre-approval checklist ##

Remove any of the following that do not apply. If you're unsure about
any of these, don't hesitate to ask. We're here to help!

- [x] ✌️ The conflicts in this pull request have been resolved.
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

---

> [!NOTE]
> You are seeing this because one of this repository's maintainers has
> configured [Lineage] to open pull requests.

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by jsf9k — CHANGES_REQUESTED** [review]
If we want to keep this repo in sync with the recent changes in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) then I think:
- `terraform` should be bumped to version 1.1 in `versions.tf`.  This would require an update to the `README.md` file as well.
- All Terraform variables should include `nullable = false` (or `nullable = true`, if appropriate)

See, e.g., cisagov/cool-userservices-dns#35.

**Comment by dav3r** [comment]
> If we want to keep this repo in sync with the recent changes in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) then I think:
> 
> * `terraform` should be bumped to version 1.1 in `versions.tf`.  This would require an update to the `README.md` file as well.
> * All Terraform variables should include `nullable = false` (or `nullable = true`, if appropriate)
> 
> See, e.g., [cisagov/cool-userservices-dns#35](https://github.com/cisagov/cool-userservices-dns/pull/35).

@mcdonnnj Do you have any concerns or issues with me making the changes that @jsf9k mentioned above?

**Comment by mcdonnnj** [comment]
> If we want to keep this repo in sync with the recent changes in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) then I think:
> 
>     * `terraform` should be bumped to version 1.1 in `versions.tf`.  This would require an update to the `README.md` file as well.
> 
>     * All Terraform variables should include `nullable = false` (or `nullable = true`, if appropriate)
> 
> 
> See, e.g., [cisagov/cool-userservices-dns#35](https://github.com/cisagov/cool-userservices-dns/pull/35).

Also @dav3r. Those changes are fine, but I think they should be made separate to this PR.

**Comment by dav3r** [comment]
> > If we want to keep this repo in sync with the recent changes in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) then I think:
> > ```
> > * `terraform` should be bumped to version 1.1 in `versions.tf`.  This would require an update to the `README.md` file as well.
> > 
> > * All Terraform variables should include `nullable = false` (or `nullable = true`, if appropriate)
> > ```
> >   
> > See, e.g., [cisagov/cool-userservices-dns#35](https://github.com/cisagov/cool-userservices-dns/pull/35).
> 
> Also @dav3r. Those changes are fine, but I think they should be made separate to this PR.

@jsf9k I just chatted with @mcdonnnj about this and I agree with him.  Since that `nullable` change is from `skeleton-tf-module` and this repo descends from `skeleton-generic`, it makes sense to put those `nullable` changes in a separate PR.  I can take care of that.  That being said, both @mcdonnnj and I think that this PR is now good to go.

**Review by dav3r — APPROVED** [review]

**Review by jsf9k — APPROVED** [review]

**Comment by jsf9k** [comment]
> > > If we want to keep this repo in sync with the recent changes in [cisagov/skeleton-tf-module](https://github.com/cisagov/skeleton-tf-module) then I think:
> > > ```
> > > * `terraform` should be bumped to version 1.1 in `versions.tf`.  This would require an update to the `README.md` file as well.
> > > 
> > > * All Terraform variables should include `nullable = false` (or `nullable = true`, if appropriate)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > See, e.g., [cisagov/cool-userservices-dns#35](https://github.com/cisagov/cool-userservices-dns/pull/35).
> > 
> > 
> > Also @dav3r. Those changes are fine, but I think they should be made separate to this PR.
> 
> @jsf9k I just chatted with @mcdonnnj about this and I agree with him. Since that `nullable` change is from `skeleton-tf-module` and this repo descends from `skeleton-generic`, it makes sense to put those `nullable` changes in a separate PR. I can take care of that. That being said, both @mcdonnnj and I think that this PR is now good to go.

See #843.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -177,17 +186,27 @@ repos:
...
+          # Note that the pip-audit pre-commit hook identifies a vulnerability
+          # in ansible-core 2.16.13.  Normally we would pin ansible-core
+          # accordingly (>2.16.13), but the above pin of ansible>=10 effectively
+          # pins ansible-core to >=2.17 so that's what we do here.
+          #
           # Note that any changes made to this dependency must also be
           # made in requirements.txt in cisagov/skeleton-packer and
           # requirements-test.txt in cisagov/skeleton-ansible-role.
-          - ansible-core>=2.16.7
+          - ansible-core>=2.17
```
Comment (mcdonnnj) [inline .pre-commit-config.yaml]: @dav3r Why was this changed? I don't believe this change should be made until https://github.com/cisagov/skeleton-generic/pull/196 filters down. This change has not been made in https://github.com/cisagov/skeleton-ansible-role/pull/206 or https://github.com/cisagov/skeleton-ansible-role/pull/209 either.

File: requirements.txt [hunk 2]
```diff
@@ -8,24 +8,29 @@
...
+# We have tested against version 10.  We want to avoid automatically
 # jumping to another major version without testing, since there are
 # often breaking changes across major versions.  This is the reason
 # for the upper bound.
-#
-# Note that this dependency should be kept in sync with the version
-# pin found in the requirements-test.txt file in
-# cisagov/skeleton-ansible-role.
-ansible>=9,<10
+ansible>=10,<11
```
Comment (mcdonnnj) [inline requirements.txt:19]: Not a fan of these changes being roped into this pull request. Just like the `nullable` change they should be a separate PR in my opinion. We may as well just ignore the vulnerability flagged by the existing version since we're already ignoring vulnerabilities anyway. It can then be removed when the version is changed to a non-vulnerable version.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dav3r) [reply .pre-commit-config.yaml]: I didn't want to spend any more time waiting for this repo to get to the state where it will be as soon as those skeleton PRs are merged.  Since this change is going to be included an upcoming lineage PR anyway (when https://github.com/cisagov/skeleton-generic/pull/196 is merged), I figured we might as well save ourselves some time and get this repo in a better place sooner rather than later.

File: requirements.txt (hunk 2 above)
Comment (dav3r) [reply requirements.txt:19]: Noted, but the commits will be the same- in my opinion, it's a waste of effort to move them around into different PRs at this point.

I thought from our earlier conversations that you were on board with just "fast forwarding" these changes in this PR, so I apologize for that misunderstanding.  But since we are here now, I just want to get this PR merged as is so that I can take care of the changes requested in https://github.com/cisagov/cyhy-system/issues/129.  Thanks!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dav3r) [reply .pre-commit-config.yaml]: https://github.com/cisagov/skeleton-generic/pull/196 has been merged - can we please resolve this conversation?

File: requirements.txt (hunk 2 above)
Comment (dav3r) [reply requirements.txt:19]: @mcdonnnj Anything else or can we now resolve this conversation?
